app/vmstorage/main.go

```diff
@@ -521,12 +521,22 @@ func writeStorageMetrics(w io.Writer, strg *storage.Storage) {
 	metrics.WriteGaugeUint64(w, `vm_data_size_bytes{type="indexdb/inmemory"}`, idbm.InmemorySizeBytes)
 	metrics.WriteGaugeUint64(w, `vm_data_size_bytes{type="indexdb/file"}`, idbm.FileSizeBytes)
 
+	metrics.WriteCounterUint64(w, `vm_rows_received_by_storage_total`, m.RowsReceivedTotal)
```

Comment:
Seems like this metric is already represented in form of `vm_vminsert_metrics_read_total`

Reply:
This metic is defined in [cluster branch](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/91d989d2d16195f42095bd834f884b0766524151/app/vmstorage/servers/vminsert.go#L143).
In VM we have two branches: master and cluster. Where master is for single-node and cluster is for clustered version. They share about 95% of the source code, but still contain some cluster-related differences.

Reply:
I see. But it feels like `vm_vminsert_metrics_read_total` count represents the number of records sent to the cluster as a whole. While the new metric will count the number of received metrics per replica. I.e. a single replica would have all the counts in one place so the dropped records could be diagnosed more conveniently.